backend/debug/dwarf/dwarf_ocaml/dwarf_concrete_instances.ml

```diff
@@ -37,31 +37,37 @@ let for_fundecl ~get_file_id state fundecl =
   in
   let start_sym = Asm_symbol.create fun_name in
   let end_sym = Asm_symbol.create (end_symbol_name ~start_symbol:fun_name) in
-  let file_num, line, startchar =
+  let location_attributes =
     let loc = Debuginfo.to_location fundecl.fun_dbg in
     if Location.is_none loc
-    then 0, 0, 0
+    then [DAH.create_artificial ()]
     else
       let file, line, startchar = Location.get_pos_info loc.loc_start in
-      get_file_id file, line, startchar
+      let attributes = [DAH.create_decl_file (get_file_id file)] in
+      let attributes =
+        if line > 0 then
+          (DAH.create_decl_line line) :: attributes
+        else attributes
+      in
+      if startchar > 0 then
+        (DAH.create_decl_column startchar) :: attributes
```

Comment:
I don't think we should emit the column attribute if there isn't a valid number, since it's likely meaningless.